src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs

```diff
@@ -114,51 +116,53 @@ public override bool Execute()
 
                     using (HttpResponseMessage response = await client.SendAsync(req))
                     {
-                        this.Log.LogMessage(
-                            MessageImportance.Normal,
+                        Log.LogMessage(
+                            MessageImportance.Low,
                             "Received response to create Container {0}: Status Code: {1} {2}",
-                            this.ContainerName, response.StatusCode, response.Content);
+                            ContainerName, response.StatusCode, response.Content.ToString());
+
+                        if (response.IsSuccessStatusCode)
+                        {
+                            try
+                            {
+                                // specifying zero is valid, it means "I don't want a token"
+                                if (ReadOnlyTokenDaysValid > 0)
+                                {
+                                    ReadOnlyToken = AzureHelper.CreateContainerSasToken(
+                                        AccountName,
+                                        ContainerName,
+                                        AccountKey,
+                                        AzureHelper.SasAccessType.Read,
+                                        ReadOnlyTokenDaysValid);
+                                }
+
+                                // specifying zero is valid, it means "I don't want a token"
+                                if (WriteOnlyTokenDaysValid > 0)
+                                {
+                                    WriteOnlyToken = AzureHelper.CreateContainerSasToken(
+                                        AccountName,
+                                        ContainerName,
+                                        AccountKey,
+                                        AzureHelper.SasAccessType.Write,
+                                        WriteOnlyTokenDaysValid);
+                                }
+
+                                success = true;
+                            }
+                            catch (ArgumentOutOfRangeException e)
+                            {
+                                Log.LogErrorFromException(e, true);
+                            }
+                        }
+                        else
+                        {
+                            Log.LogError("Container creation failed with response {0}", response.StatusCode);
```

Comment:
This is where it will end up if the container name is bogus right?  Might be worth putting in a check for anything that isn't alphanumeric lower case or - here, so the error is more prescriptive.